projects/mercury/src/llm/Conversation.js

```diff
@@ -0,0 +1,415 @@
+import React, {useEffect, useState} from 'react';
+import {
+    Box,
+    Button,
+    CircularProgress,
+    Fade,
+    Grid,
+    IconButton,
+    List,
+    ListItemButton,
+    Modal,
+    Paper,
+    TextField,
+    Tooltip,
+    Typography
+} from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
+import CloseIcon from '@mui/icons-material/Close';
+import SearchIcon from '@mui/icons-material/Search';
+import withStyles from '@mui/styles/withStyles';
+import FulltextAPI from './FulltextAPI';
+import styles from './Conversation.styles';
+import {handleHttpError} from '../common/utils/httpUtils';
+import LinkedDataEntityPage from '../metadata/common/LinkedDataEntityPage';
+import {LocalSearchAPI} from '../search/SearchAPI';
+
+const Conversation = props => {
```

Comment:
This huge cla should be divided into smaller modules, currently it is hard to follow what happens here